qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -1,18 +1,332 @@
 import numpy as np
 import qcodes.utils.validators as vals
+from qcodes import InstrumentChannel, MultiParameter
+from qcodes.instrument.group_parameter import GroupParameter, Group
 from qcodes.utils.validators import Arrays
 
 from .KeysightB1500_sampling_measurement import SamplingMeasurement
-from .KeysightB1500_module import B1500Module, parse_spot_measurement_response
+from .KeysightB1500_module import B1500Module, \
+    parse_spot_measurement_response, parse_fmt_1_0_response, _FMTResponse
 from .message_builder import MessageBuilder
 from . import constants
-from .constants import ModuleKind, ChNr, AAD, MM
+from .constants import ModuleKind, ChNr, AAD, MM, IMeasRange
+
 if TYPE_CHECKING:
     from .KeysightB1500_base import KeysightB1500
 
 
+class IVSweeper(InstrumentChannel):
+    def __init__(self, parent: 'B1517A', name: str, **kwargs: Any):
+        super().__init__(parent, name, **kwargs)
+
+        self.add_parameter(name='sweep_auto_abort',
+                           set_cmd=self._set_sweep_auto_abort,
+                           set_parser=constants.Abort,
+                           vals=vals.Enum(*list(constants.Abort)),
+                           get_cmd=None,
+                           docstring=textwrap.dedent("""
+        The WM command enables or disables the automatic abort function for 
+        the staircase sweep sources and the pulsed sweep source. The 
+        automatic abort function stops the measurement when one of the 
+        following conditions occurs:
+         - Compliance on the measurement channel
+         - Compliance on the non-measurement channel
+         - Overflow on the AD converter
+         - Oscillation on any channel
+        This command also sets the post measurement condition for the sweep 
+        sources. After the measurement is normally completed, the staircase 
+        sweep sources force the value specified by the post parameter, 
+        and the pulsed sweep source forces the pulse base value.
+        
+        If the measurement is stopped by the automatic abort function, 
+        the staircase sweep sources force the start value, and the pulsed 
+        sweep source forces the pulse base value after sweep.
+        """))
+        self.sweep_auto_abort.cache.set(constants.Abort.ENABLED)
+
+        self.add_parameter(name='post_sweep_voltage_condition',
+                           set_cmd=self._set_post_sweep_voltage_condition,
+                           set_parser=constants.WM.Post,
+                           vals=vals.Enum(*list(constants.WM.Post)),
+                           get_cmd=None,
+                           docstring=textwrap.dedent("""
+        Source output value after the measurement is normally completed. If 
+        this parameter is not set, the sweep sources force the start value.
+                                 """))
+        self.post_sweep_voltage_condition.cache.set(constants.WM.Post.START)
```

Comment:
We now have `initial_cache_value` for this (as an arg to add_parameter)

Reply:
done